internal/mode/static/handler_test.go

```diff
@@ -275,19 +278,137 @@ var _ = Describe("eventHandler", func() {
 			}
 			batch := []interface{}{e}
 
-			handler.HandleEventBatch(context.Background(), ctlrZap.New(), batch)
+			handler.HandleEventBatch(context.Background(), batch)
 			Expect(fakeStatusUpdater.UpdateAddressesCallCount()).ToNot(BeZero())
 		})
 	})
 
+	When("receiving an EndpointsOnlyChange update", func() {
+		e := &events.UpsertEvent{Resource: &discoveryV1.EndpointSlice{
```

Comment:
In EndpointsOnlyChange  tests, shall we also ensure that we call ReplaceFiles? to ensure that the config files are written too